src/resolver/fileLookup.ts

```diff
@@ -100,43 +102,56 @@ export function fileLookup(props: ILookupProps): ILookupResult {
   const exists = fileExists(resolved);
   if (exists) {
     const stat = fs.lstatSync(resolved);
-    if (stat.isDirectory) {
-      isDirectory = true;
-
-      let monorepoModulesPaths;
-      let tsConfigAtPath: TsConfigAtPath;
-
-      // only in case of a directory
-      const packageJSONPath = path.join(resolved, 'package.json');
-      if (fileExists(packageJSONPath)) {
-        const useLocalMain = !/node_modules/.test(packageJSONPath);
-        const packageJSON = require(packageJSONPath);
-        const entry = getFolderEntryPointFromPackageJSON({ json: packageJSON, useLocalField: useLocalMain });
-
-        if (useLocalMain && packageJSON['local:main']) {
-          const _monoModules = path.resolve(resolved, 'node_modules');
-          if (fileExists(_monoModules)) {
-            monorepoModulesPaths = _monoModules;
+    isDirectory = stat.isDirectory();
+
+    if (isDirectory) {
+      if (props.inBaseUrl !== true) {
+        let monorepoModulesPaths;
+        let tsConfigAtPath: TsConfigAtPath;
+        // only in case of a directory
+        const projectRoot = path.join(resolved, "../../../"); // TODO: get the actual projectRoot
+        const packageJSONPath = findUp(resolved, 'package.json', {
+          boundary: projectRoot,
+          inclusive: false,
+        });
+        const packageRoot = path.dirname(packageJSONPath)
+        if (fileExists(packageJSONPath)) {
+          const useLocalMain = !/node_modules/.test(packageJSONPath);
+          const packageJSON = require(packageJSONPath);
+          const entry = getFolderEntryPointFromPackageJSON({ json: packageJSON, useLocalField: useLocalMain });
+
+          if (packageJSON['main:local']) {
```

Comment:
Do we need this check? xD ;-))

Reply:
Silly as it is, I ended up banging my head for a while before I realized I'd inverted the order of these keywords.

I'd rather find a way to check if something really looks like it should be a "sibling module" and give a small notice if there is no "local:main".